.github/workflows/react.yml

```diff
@@ -0,0 +1,45 @@
+name: React
+
+on:
+  push:
+    branches: [ "main" ]
+    paths:
+      - 'frontend/**'
+  pull_request:
+    branches: [ "main" ]
+    paths:
+      - 'frontend/**'
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20
+
+      - name: Cache node_modules
+        uses: actions/cache@v3
+        with:
+          path: frontend/node_modules
+          key: ${{ runner.os }}-node-${{ hashFiles('frontend/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        working-directory: frontend
+        run: npm install
+
+      - name: Run build
+        working-directory: frontend
+        run: npm run build
+#
```

Comment:
No tests yet? Will this fail if we have no tests defined? If it will pass I think this should be uncommented

Reply:
Right now this won't work because we don't have tesing library. I will work on this later.

Reply:
Done